graphql_compiler/tests/test_compiler.py

```diff
@@ -5423,6 +5423,74 @@ def test_fold_on_many_to_one_edge(self):
             expected_postgresql,
         )
 
+    def test_fold_after_recurse(self):
+        test_data = test_input_data.fold_after_recurse()
+
+        expected_postgresql = SKIP_TEST
+        expected_mssql = """
+        WITH anon_1(lives_in, parent, uuid, __cte_key, __cte_depth) AS (
+            SELECT
+                [Animal_1].lives_in AS lives_in,
+                [Animal_1].parent AS parent,
+                [Animal_1].uuid AS uuid,
+                [Animal_1].uuid AS __cte_key,
+                0 AS __cte_depth
+            FROM
+                db_1.schema_1.[Animal] AS [Animal_1]
+            UNION ALL
+            SELECT
+                [Animal_2].lives_in AS lives_in,
+                [Animal_2].parent AS parent,
+                [Animal_2].uuid AS uuid,
+                anon_1.__cte_key AS __cte_key,
+                anon_1.__cte_depth + 1 AS __cte_depth
+            FROM
+                anon_1 JOIN db_1.schema_1.[Animal] AS [Animal_2]
+                    ON anon_1.uuid = [Animal_2].parent
+            WHERE
+                anon_1.__cte_depth < 3
+        )
+        SELECT
+            [Animal_1].name AS animal_name,
+            folded_subquery_1.fold_output_name AS homes_list
+        FROM
+            db_1.schema_1.[Animal] AS [Animal_1],
+            anon_1 JOIN (
+                SELECT
+                    anon_2.uuid AS uuid,
+                    coalesce((
+                        SELECT
+                            '|' + coalesce(
+                                REPLACE(
+                                    REPLACE(
+                                        REPLACE([Location_1].name, '^', '^e'),
+                                    '~',
+                                    '^n'),
+                                '|',
+                                '^d'),
+                            '~')
+                        FROM
+                            db_1.schema_1.[Location] AS [Location_1]
+                        WHERE
+                            anon_2.lives_in = [Location_1].uuid FOR XML PATH ('')),
+                   '') AS fold_output_name
+                FROM anon_1 AS anon_2
+            ) AS folded_subquery_1
+                ON anon_1.uuid = folded_subquery_1.uuid
+        """
+        expected_match = SKIP_TEST
+        expected_gremlin = SKIP_TEST
+        expected_cypher = SKIP_TEST
```

Comment:
When do we decide to use `SKIP_TEST` vs `NotImplementedError`/emitted query? If this test is really only relevant for MSSQL, might be nice to include a note about why that is.

Reply:
`SKIP_TEST`: Either not implemented yet, not tested yet, or no reason to test.
`NotImplementedError`: The desired behavior is to return `NotImplementedError`, and that's implemented.

I'll add a comment.

Reply:
This is one of the things I'd love to eradicate in a future revamp of our test suite. "Not tested yet / no reason to test" situations shouldn't need explicit hand-written expected output, but if we have the test case inputs ready to go, we could definitely auto-generate a snapshot of the current behavior and alert if it ever changes. It would be a nice defense-in-depth strategy, but only worth it if we can make it easier to do.

Right now, if writing a test targeting something that's generally only interesting e.g. for a particular SQL flavor (e.g. `@fold` in MSSQL), we either have to write the expected test outcome for other SQLs and other query languages (tedious!) or `SKIP_TEST` to reduce the amount of effort a PR takes and not take advantage of the additional test opportunity. This is where an automatic snapshot test would be useful: low effort yet still helpful.